### PR TITLE
Add bearer token authentication documentation to template

### LINKED ISSUE #4178: [BUG][Java] isBasicBearer not used in default template
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

Generating the code with the default Java template, it seems the new added `isBasicBearer` is not used:
https://github.com/OpenAPITools/openapi-generator/blob/7ab73ff587a4c112307f9b646d4af744bb17d152/modules/openapi-generator/src/main/resources/Java/ApiClient.mustache#L106-L108

The Bearer part is defined inside the `isBasicBasic`. I thought it would be within the `isBasicBearer`.

##### openapi-generator version

4.1.3

##### OpenAPI declaration file content or url

Although the problem is at the template level, not the code generation I think, here is the link to the OpenAPI spec:
https://api.criteo.com/marketing/swagger/docs/v.1.0

##### Command line used for generation
https://github.com/criteo/criteo-marketing-sdk-generator/blob/master/build-java/build.gradle#L66-L93

Thanks :)

### PR DESCRIPTION
**Pull request by addynaik**
The Readme files generated  for java currently only adds HttpBasicAuth example. This update checks if the auth is a bearer auth and adds appropriate example for that. 

This fixes #4178

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by addynaik** [comment]
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04)

File: modules/openapi-generator/src/main/resources/Java/README.mustache [hunk 1]
```diff
@@ -89,11 +89,14 @@ public class {{{classname}}}Example {
     public static void main(String[] args) {
         ApiClient defaultClient = Configuration.getDefaultApiClient();
         defaultClient.setBasePath("{{{basePath}}}");
-        {{#hasAuthMethods}}{{#authMethods}}{{#isBasic}}
+        {{#hasAuthMethods}}{{#authMethods}}{{#isBasic}}{{#isBasicBasic}}
         // Configure HTTP basic authorization: {{{name}}}
-        HttpBasicAuth {{{name}}} = (HttpBasicAuth) defaultClient.getAuthentication("{{{name}}}");
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/README.mustache]: @addynaik may I know why "{{{name}}}" is removed from `getAuthentication()`?

Comment (addynaik) [reply modules/openapi-generator/src/main/resources/Java/README.mustache]: it should not be.. i think that was removed by accident. Thanks for catching that.

Comment (addynaik) [reply modules/openapi-generator/src/main/resources/Java/README.mustache]: @wing328 this should be fixed now

**Pull request merged**